### PR TITLE
Feat: create paginator template + test with Recipe Embeds + handle edge cases of finding recipes

### PR DESCRIPTION
**Pull request by xinyue296**
1. Created a paginator template for displaying embeds because:
    - discord allows up to 10 embeds per message, but with a paginator, we can handle more
    - sending 10 separate embeds takes up a lot of space and is hard to follow
2. Managed to test the paginator template with Recipe Embeds
3. Handled edge cases in finding recipes to ensure the paginator handles scenarios where a user requests:
    - a negative number of recipes
    - more than 20 recipes per call
4. Grouped files by user case within existing packages
5. Added test files for the recipe feature
6. Fixed typo in `build.gradle.kts`

### CODE REVIEW

**Review by Overpoweredplatypus — APPROVED** [review]
I am so happy for you!

**Review by sailoraria — APPROVED** [review]
This is such good work thank you!

**Comment by iDasher** [comment]
Looks good!

**Review by iDasher — APPROVED** [review]
